packages/forms/src/widgets/DatalistWidget/DatalistWidget.js

```diff
@@ -261,39 +254,122 @@ class DatalistWidget extends React.Component {
 
 	selectItem(itemIndex) {
 		const selectedItem = this.state.items[itemIndex];
-		if (selectedItem && selectedItem !== this.state.value) {
+		const selectedItemLabel = this.getLabel(selectedItem);
+
+		if (selectedItemLabel && selectedItemLabel !== this.state.value) {
 			this.setValue(selectedItem);
 			this.resetSuggestions();
 			this.props.onChange(selectedItem);
 		}
 	}
 
+	/**
+	 * Render the datalist suggestion item
+	 * @param item
+	 * @param value
+	 */
+	renderDatalistItem(item, { value }) {
+		const label = this.getLabel(item);
+		let emphasisedText = [label];
+
+		if (value) {
+			emphasisedText = [];
+			const regex = new RegExp(escapeRegexCharacters(value), 'gi');
+			const matchedValues = label.match(regex);
+			const restValues = label.split(regex);
+
+			restValues.forEach((val, i) => {
+				emphasisedText.push(val);
+				if (matchedValues[i]) {
+					emphasisedText.push(<em className={theme['highlight-match']}>{matchedValues[i]}</em>);
+				}
+			});
+		}
+
+		return (
+			<div className={classnames(theme.item, 'datalist-item')}>
+				{emphasisedText.map((val, index) => <span key={index}>{val}</span>)}
+			</div>
+		);
+	}
+
+	/**
+	 * Render the datalist input
+	 * @param props
+	 */
+	renderDatalistInput(props) {
+		return (
+			<div className={theme['typeahead-input-icon']}>
+				<FormControl
+					{...props}
+					value={this.getLabel(props.value)}
+				/>
+				<div className={theme['dropdown-toggle']}>
+					<span className="caret" />
+				</div>
+			</div>);
+	}
+
 	render() {
-		const renderItemData = { value: this.state.value };
-		this.inputProps.value = this.state.value;
-		const renderItemsContainer =
-			this.props.renderItemsContainer || defaultRenderDatalistItemContainer;
+		let renderItemsContainer;
+		const value = this.state.value;
+		const renderItemData = { value };
 		const renderNoMatch = this.props.renderNoMatch || defaultRenderNoMatch;
+
+		this.inputProps.value = value;
+		if (this.state.noMatch) {
+			renderItemsContainer = renderNoMatch;
```

Comment:
It can't without doing what it was done before this little refactoring. Before, we had :

```javascript
const renderItemsContainer = this.props.renderItemsContainer || defaultRenderDatalistItemContainer;
const renderNoMatch = this.props.renderNoMatch || defaultRenderNoMatch;

...

renderItemsContainer={this.state.noMatch ? renderNoMatch : renderItemsContainer}
```

I think it is clearer with this single if/else. What do you think about that ?